### PR TITLE
[Autotune] Add SizedMultiSet

### PR DESCRIPTION
**Pull request by zhhsplendid**
Pre-work for auto tuning.

The sized multi set will be used to store auto tuning records.

### CODE REVIEW

File: cinn/utils/sized_multi_set.h [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    }
+  }
+
+  void Pop() {
+    if (pop_max_when_full_) {
+      multi_set_.erase(--multi_set_.end());
+    } else {
+      multi_set_.erase(multi_set_.begin());
+    }
+  }
```
Comment (CtfGo) [inline cinn/utils/sized_multi_set.h]: set为空时erase会抛异常吗

Comment (zhhsplendid) [reply cinn/utils/sized_multi_set.h]: 会，不过是stl抛的异常，我这边不处理。

Comment (CtfGo) [reply cinn/utils/sized_multi_set.h]: 那是否为空时直接返回，或者重抛异常给用户提示

**Review by CtfGo — APPROVED** [review]
LGTM

**Pull request merged**